config/base/src/lib.rs

```diff
@@ -256,3 +271,59 @@ pub mod view {
         pub const IS_HAS_VIEW: bool = true;
     }
 }
+
+pub mod proxy {
+    //! Module for configuration proxies' traits
+
+    use super::*;
+
+    /// Pseudo-default trait only used for doc generation
+    pub trait DocsDefault {
+        fn default() -> Self;
+    }
+
+    /// Trait used to convert configs from file and env
+    pub trait Combine: Sized + Serialize + DeserializeOwned {
+        /// Which type of [`Configuration`] it builds into
+        type Target;
+
+        /// Build the config, do the necessary checks
+        fn build(self) -> Result<Self::Target, derive::Error>;
+
+        /// If any of the fields in [`other`] are filled, they
+        /// override the values of the fields in [`self`].
+        fn combine(self, other: Self) -> eyre::Result<Self, eyre::Error>;
+
+        /// Construct [`Self`] from a path-like object.
+        ///
+        /// # Errors
+        /// - File not found.
+        /// - File found, but peer configuration parsing failed.
+        /// - The length of the array in raw JSON representation is different
+        /// from the length of the array in
+        /// [`self.sumeragi.trusted_peers.peers`], most likely due to two
+        /// (or more) peers having the same public key.
+        fn from_path<P: AsRef<Path> + Debug + Clone>(path: P) -> eyre::Result<Self, eyre::Error> {
+            let file =
+                File::open(&path).wrap_err(format!("Failed to open the config file {:?}", path))?;
+            let reader = BufReader::new(file);
+            serde_json::from_reader(reader)
+                .wrap_err(format!("Failed to deserialize json {:?} from reader", path))
+        }
+
+        // fn finalize(&mut self) -> Result<()> {
+        //     self.sumeragi.key_pair =
+        //         KeyPair::new(self.public_key.clone(), self.private_key.clone())?;
+        //     self.sumeragi.peer_id =
+        //         iroha_data_model::peer::Id::new(&self.torii.p2p_addr, &self.public_key.clone());
+
+        //     Ok(())
+        // }
+
+        /// Load configuration from the environment
+        ///
+        /// # Errors
+        /// Fails if Configuration deserialization fails (e.g. if `TrustedPeers` contains entries with duplicate public keys)
+        fn load_environment(&mut self) -> core::result::Result<(), derive::Error>;
```

Comment:
Again, separate trait. These traits should be pre-requisites for something to be configurable, not part of the API. 

Reason being that we might want to have stuff that we can load from the ENV, but not have it part of config.